### DISCUSSION TITLE
Add custom page to main route

### DISCUSSION THREAD

**Discussion by iamvaliyev** [directus/directus#3293] [Archive]
I need to add a landing page to a project how to do it?

**Comment by danieljwestman** [comment]
Tip: check out Discord post by @rijkvanzanten 

https://discord.com/channels/725371605378924594/741330910594465823/781528954121093140

Fun fact (and this is very experimental still, and not documented yet because of that), buuuut:

The Directus express app is exported in the `directus` package too, so you should be able to add `directus` as an endpoint of another express (compatible) based project!

So you can do something like

```js
const express = require('express');
const directus = require('directus/dist/app');

const app = express()
  .get('/directus', directus)
  .get('/', (req, res) => res.render('my-app'))
  .listen(8080);
```

**Comment by iamvaliyev** [comment]
> Tip: check out Discord post by @rijkvanzanten
> 
> https://discord.com/channels/725371605378924594/741330910594465823/781528954121093140
> 
> Fun fact (and this is very experimental still, and not documented yet because of that), buuuut:
> 
> The Directus express app is exported in the `directus` package too, so you should be able to add `directus` as an endpoint of another express (compatible) based project!
> 
> So you can do something like
> 
> ```js
> const express = require('express');
> const directus = require('directus/dist/app');
> 
> const app = express()
>   .get('/directus', directus)
>   .get('/', (req, res) => res.render('my-app'))
>   .listen(8080);
> ```

I get error : **Error: Route.get() requires a callback function but got a [object Object]**

**Reply by rijkvanzanten** [reply]
That's where the "So you can do something _like_" part comes in

Seeing we use TS w/ esm exports, you might have to use 

```
const directus = require('directus/dist/app').default;
```

instead, and it's

`express.use('/directus', directus)`, not `.get`

**Reply by customautosys** [reply]
I am trying to integrate directus into a Quasar framework SSR app but it is not working:

```typescript
import {ssrMiddleware} from 'quasar/wrappers';
import {createApp} from 'directus';

export default ssrMiddleware(({app})=>app.use('/directus',createApp));
```

Any ideas? Basically the app object is just an express server, normally it just works.

Basically what happens is that the request just hangs and never returns. I know it is calling directus because I was getting a PostGIS not installed error as well as a PUBLIC_URL error until I fixed those.

**Reply by emahuni** [reply]
did you manage to integrate with Quasar? and how?

**Reply by customautosys** [reply]
> did you manage to integrate with Quasar? and how?

Yes, see this thing I created:

https://www.npmjs.com/package/directus-transport-safe-run-middleware

However, I then got stuck because Directus 9 was unable to run on Namecheap due to some native package incompatibilities with the old CloudLinux version. So in the end, I integrated my Quasar site SSR with my existing Directus 8 PHP backend using the old JS SDK instead: https://customautosys.com
